core/src/main/java/org/mule/runtime/core/routing/AbstractAggregator.java

```diff
@@ -73,8 +73,8 @@
   @Override
   public void initialise() throws InitialisationException {
     if (storePrefix == null) {
-      storePrefix =
-          format("%s%s.%s.", muleContext.getConfiguration().getId(), flowConstruct.getName(), this.getClass().getName());
+      storePrefix = format("%s%s.%s.", muleContext.getConfiguration().getId(), getLocation().getParts().get(0).getPartPath(),
```

Comment:
Refactor repeated use of this type of thing into static helpers in ComponentLocation.